### DISCUSSION TITLE
Repeat a pattern over a very large image

### DISCUSSION THREAD

**Discussion by explorer-vivek** [libvips/libvips#3394] [Q&A]
I need to repeat a pattern image over a very large image. For eg - The input pattern image can be an 600x600 px PNG file and I need to create a new very large PNG (128,000 x 128,000 px) such that it contains the input pattern repeated across the x and y directions. Is it possible to do this using libvips?

**Comment by jcupitt** [comment]
Hi @explorer-vivek, 

Sure, use `replicate` to copy an image many times. 

https://www.libvips.org/API/current/libvips-conversion.html#vips-replicate

For example:

```
$ vipsheader PNG_transparency_demonstration.png 
PNG_transparency_demonstration.png: 800x600 uchar, 4 bands, srgb, pngload
$ time vips replicate PNG_transparency_demonstration.png x.png 160 215

real	4m14.495s
user	5m3.268s
sys	0m16.394s
$ vipsheader x.png
x.png: 128000x129000 uchar, 4 bands, srgb, pngload
$ ls -l x.png
-rw-r--r-- 1 john john 287538028 Mar 20 12:31 x.png
```

Be aware that large PNGs like this are very slow to process. You'll probably find that something like TIFF is easier to handle.

**Reply by explorer-vivek** [reply]
@jcupitt Thanks for pointing me to the correct API. It is indeed very slow for large PNG files. 
Also, what are the possible options that can be passed to the `replicate` API?

**Reply by jcupitt** [reply]
The docs list all the options (there aren't any heh). What do you need to do?
